src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -753,6 +759,38 @@ public static ServerHostInfo GetServerHostInfo(IDbConnection connection)
             return hostInfo;
         }
 
+        /// <summary>
+        /// Gets the server host cpu count and memory from sys.dm_os_sys_info view
+        /// </summary>
+        /// <param name="connection">The connection</param>
+        public static ServerSysInfo GetServerSysInfo(IDbConnection connection)
+        {
+            var sysInfo = new ServerSysInfo();
+            try
+            {
+               SqlConnection conn = null;
+               ReliableSqlConnection reliableSqlConnection = connection as ReliableSqlConnection;
+               SqlConnection sqlConnection = connection as SqlConnection;
+                if (reliableSqlConnection != null)
+                {
+                    conn = reliableSqlConnection.GetUnderlyingConnection();
+                }
+                else if (sqlConnection != null)
+                {
+                    conn = sqlConnection;
+                }
+                var server = new Server(new ServerConnection(conn));
+                server.SetDefaultInitFields(server.GetType(), new String[] { "Processor", "PhysicalMemory" });
```

Comment:
use nameof(server.Processor), nameof(server.PhysicalMemory) instead of string constants